source/read/cursors.txt

```diff
@@ -0,0 +1,104 @@
+.. _java-rs-cursors:
+
+=========================
+Access Data From a Cursor
+=========================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: read, results, oplog
+
+Overview
+--------
+
+In this guide, you can learn how to access data from a **cursor** with the
+{+driver-short+}.
+
+A cursor is a mechanism that returns the results of a read operation in iterable
+batches. Because a cursor holds only a subset of documents at any given time,
+cursors reduce both memory consumption and network bandwidth usage.
+
+Whenever the {+driver-short+} performs a read operation that returns multiple
+documents, it automatically returns those documents in a cursor.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``sample_restaurants.restaurants`` collection
+from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+free MongoDB Atlas cluster and load the sample datasets, see the
+:ref:`<java-rs-getting-started>` guide.
+
+.. include:: includes/reactor-note.rst
+
+Access Cursor Contents Iteratively
+----------------------------------
+
+To iterate over the contents of a cursor, use the ``Flux.from()`` method, as shown in the
+following example:
+
+.. literalinclude:: /includes/read-ops/cursors.java
+   :start-after: start-cursor-iterate
+   :end-before: end-cursor-iterate
+   :language: java
+   :dedent:
+   :copyable:
+
+Retrieve All Documents
+----------------------
+
+.. warning::
+
+   If the number and size of documents returned by your query exceeds available
+   application memory, your program will crash. If you expect a large result
+   set, :ref:`access your cursor iteratively <kotlin-sync-cursors-iterate>`.
+
+To retrieve all documents from a cursor, convert the cursor into a ``List`` as
+shown in the following example:
+
+.. literalinclude:: /includes/read-ops/cursors.java
+   :start-after: start-cursor-list
+   :end-before: end-cursor-list
+   :language: java
+   :dedent:
+   :copyable:
+
+Tailable Cursors
+----------------
+
+When querying on a :manual:`capped collection </core/capped-collections/>`, you
+can use a **tailable cursor** that remains open after the client exhausts the
+results in a cursor. To create a tailable cursor with capped collection,
+specify ``CursorType.TailableAwait`` to the ``cursorType`` method of a
+``FindPublisher`` object.
+
+The following example creates a tailable cursor on a collection:
```

Comment:
S: since the code also prints the cursor results, I'd include that in the description
```suggestion
The following example creates a tailable cursor on a collection and prints its contents:
```